chain/jsonrpc/openapi/src/main.rs

```diff
@@ -0,0 +1,563 @@
+use okapi::openapi3::{OpenApi, SchemaObject};
+use schemars::JsonSchema;
+use schemars::transform::transform_subschemas;
+use serde_json::json;
+
+use near_jsonrpc_primitives::types::{
+    blocks::{RpcBlockRequest, RpcBlockResponse},
+    changes::{
+        RpcStateChangesInBlockByTypeRequest, RpcStateChangesInBlockByTypeResponse,
+        RpcStateChangesInBlockRequest,
+    },
+    chunks::{RpcChunkRequest, RpcChunkResponse},
+    client_config::RpcClientConfigResponse,
+    config::{RpcProtocolConfigRequest, RpcProtocolConfigResponse},
+    congestion::{RpcCongestionLevelRequest, RpcCongestionLevelResponse},
+    gas_price::{RpcGasPriceRequest, RpcGasPriceResponse},
+    light_client::{
+        RpcLightClientBlockProofRequest, RpcLightClientBlockProofResponse,
+        RpcLightClientExecutionProofResponse, RpcLightClientNextBlockRequest,
+        RpcLightClientNextBlockResponse,
+    },
+    maintenance::{RpcMaintenanceWindowsRequest, RpcMaintenanceWindowsResponse},
+    network_info::RpcNetworkInfoResponse,
+    receipts::{RpcReceiptRequest, RpcReceiptResponse},
+    split_storage::{RpcSplitStorageInfoRequest, RpcSplitStorageInfoResponse},
+    status::{RpcHealthResponse, RpcStatusResponse},
+    transactions::{
+        RpcSendTransactionRequest, RpcTransactionResponse, RpcTransactionStatusRequest,
+    },
+    validator::RpcValidatorsOrderedRequest,
+    validator::{RpcValidatorRequest, RpcValidatorResponse, RpcValidatorsOrderedResponse},
+};
+use near_primitives::hash::CryptoHash;
+
+use near_chain_configs::GenesisConfig;
+
+#[cfg(not(feature = "progenitor"))]
+use near_jsonrpc_primitives::{
+    errors::RpcError,
+    types::{
+        changes::RpcStateChangesInBlockResponse, light_client::RpcLightClientExecutionProofRequest,
+    },
+};
+
+#[cfg(feature = "progenitor")]
+use near_jsonrpc_primitives::errors::RpcRequestValidationErrorKind;
+#[cfg(feature = "progenitor")]
+use serde_with::{base64::Base64, serde_as};
+
+#[derive(JsonSchema)]
+#[serde(untagged)]
+pub enum ResponseEither<T, E> {
+    Success { result: T },
+    RpcError { error: E },
+}
```

Comment:
I think this is a bit more readable, and no need for untagged:

```suggestion
pub enum ResponseEither<T, E> {
    #[serde(rename = "result")]
    Result: T,
    #[serde(rename = "error")]
    Error: E,
}
```